api/bag_layer.cpp

```diff
@@ -208,9 +208,8 @@ UInt8Array Layer::read(
     if (m_pBagDataset.expired())
         throw DatasetNotFound{};
 
-    const auto pDataset = m_pBagDataset.lock();
     uint32_t numRows = 0, numColumns = 0;
-    std::tie(numRows, numColumns) = pDataset->getDescriptor().getDims();
+    std::tie(numRows, numColumns) = m_pLayerDescriptor->getDims();
 
     if (columnEnd >= numColumns || rowEnd >= numRows)
```

Comment:
This branch is being taken when running [test_bag_vrmetadata.cpp](https://github.com/OpenNavigationSurface/BAG/blob/109-LayerDescriptor-shape-parameters/tests/test_bag_vrmetadata.cpp#L452), specifically when the test attempts to read back the VR metadata. Is the test wrong, or is there a mistake in `Layer::read()`?

Reply:
I don't see a branch here, but the code is correct --- we have to read the dimensions specifically from the layer rather than from the dataset, since we need to allow each layer to have its own dimensions.